### PR TITLE
BASH-19 Remove hardcoded product name references

### PR DESCRIPTION
**Pull request by csduarte**
**Description**
This PR changes all hardcoded product name references to use the productName from the package.json file. 

- [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

### CODE REVIEW

**Comment by yuya-oc** [comment]
I think [app.getName()](https://electron.atom.io/docs/api/app/#appgetname) is more proper in Electron.

**Comment by dmeza** [comment]
@yuya-oc rebased to latest. Replaced app.productName with app.getName() and build, packaged and run linting. Let me know if that's the correct change.

File: src/browser/settings.jsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 'use strict';
+const app = require('../package.json');
```
Comment (dmeza) [inline src/browser/settings.jsx]: @yuya-oc would your comments for use `require('../src/package.json')` apply here too?
 https://github.com/mattermost/desktop/pull/591/files#r137918302

File: src/browser/components/SettingsPage.jsx [hunk 2]
```diff
@@ -9,12 +9,13 @@ const AutoLaunch = require('auto-launch');
 const {debounce} = require('underscore');
 
 const settings = require('../../common/settings');
+const app = require('../../package.json');
```
Comment (yuya-oc) [inline src/browser/components/SettingsPage.jsx]:
```js
const {remote} = require('electron');
```

File: src/browser/components/SettingsPage.jsx [hunk 3]
```diff
@@ -9,12 +9,13 @@ const AutoLaunch = require('auto-launch');
...
 
 const TeamList = require('./TeamList.jsx');
 const AutoSaveIndicator = require('./AutoSaveIndicator.jsx');
 
 const appLauncher = new AutoLaunch({
-  name: 'Mattermost',
+  name: app.getName(),
```
Comment (yuya-oc) [inline src/browser/components/SettingsPage.jsx]: `remote.app.getName()`

File: src/browser/settings.jsx (hunk 1 above)
Comment (yuya-oc) [reply src/browser/settings.jsx]: @dmeza No. #591 is a normal node.js script, so I suggested `require('../src/package.json')`. Please use Electron API here.

File: src/browser/settings.jsx [hunk 4]
```diff
@@ -1,7 +1,8 @@
...
 
 window.eval = global.eval = () => {
-  throw new Error('Sorry, Mattermost does not support window.eval() for security reasons.');
+  throw new Error(`Sorry, ${app.getName()} does not support window.eval() for security reasons.`);
```
Comment (yuya-oc) [inline src/browser/settings.jsx]: `remote.app.getName()`

File: src/main/squirrelStartup.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
 const AutoLaunch = require('auto-launch');
+const app = require('../package.json');
```
Comment (yuya-oc) [inline src/main/squirrelStartup.js]:
```js
const {app} = require('electron');
```

**Comment by dmeza** [comment]
Thank you for the comments @yuya-oc. Rebased to latest and made the changes as you mentioned.

**Comment by yuya-oc** [comment]
@dmeza Thanks. By the way, I noticed @jarredwitt is not approved by [CLA](https://www.mattermost.org/mattermost-contributor-agreement/). His change is very small. But to make sure, I think we need his agreement. Would you confirm that?

CC @jasonblais

**Comment by jasonblais** [comment]
Agree, that would be good.

**Comment by csduarte** [comment]
I might be wrong, but jarred and dmeza are committed under myself and the company? I think dmeza only signed because he was there in the early days before we had everything setup correctly. @jasonblais At least that's what I thought was the case for mobile.

**Comment by jasonblais** [comment]
@csduarte You're correct, I just reviewed the mobile repo and this follows the same procedure. @yuya-oc The commits will be under @csduarte, 

No need to sign the CLA,

**Comment by yuya-oc** [comment]
Okay, thanks!

**Pull request merged**